pilot/pkg/model/service.go

```diff
@@ -475,8 +475,7 @@ func (ports PortList) Get(name string) (*Port, bool) {
 // GetByPort retrieves a port declaration by port value
 func (ports PortList) GetByPort(num int) (*Port, bool) {
 	for _, port := range ports {
-		if port.Port == num && port.Protocol != protocol.UDP &&
```

Comment:
@costinm do you think there is significant risk in making this change?

Some background, this PR changes `ConvertProtocol` as necessary for protocol sniffing. However, some integration tests fail without also returning `Unsupported` ports from `PortList.GetByPort`. There a few call sites for this, but I'm not sure how much it matters. Hoping you might have more context.